### PR TITLE
Fix worker pool

### PR DESCRIPTION
**Pull request by sonnysideup**
See the commit extended comments

### CODE REVIEW

File: pkg/controller/start.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func Start(cfg config.Controller) error {
 	ctrl.SetLogger(zapr.NewLogger(zapLogger))
 
 	log := ctrl.Log.WithName("setup")
-	log.V(1).Info("Using provided configuration", "config", cfg)
+	log.Info("Using provided configuration", "config", cfg)
```
Comment (steved) [inline pkg/controller/start.go:45]: This doesn't have secrets in it?

File: pkg/buildkit/worker/pool.go [hunk 2]
```diff
@@ -502,8 +520,44 @@ func (p *workerPool) triggerReconcile() {
 	}
 }
 
+// ensure pod is operational by checking its phase and conditions
+func (p *AutoscalingPool) isOperationalPod(ctx context.Context, podName string) (verdict bool) {
+	// fetch the latest version of the pod
+	pod, err := p.podClient.Get(ctx, podName, metav1.GetOptions{})
```
Comment (steved) [inline pkg/buildkit/worker/pool.go:526]: pretty minor, but should this have a resource version associated with the request (i.e. no older than the one we got from the list)?

**Review by steved — APPROVED** [review]

File: pkg/controller/start.go (hunk 1 above)
Comment (sonnysideup) [reply pkg/controller/start.go:45]: mmm, the amqp url 😬 . i'll have to figure out how logr serializes the object and make sure that field and any others get sanitized. seems like a worthwhile effort.

File: pkg/buildkit/worker/pool.go (hunk 2 above)
Comment (sonnysideup) [reply pkg/buildkit/worker/pool.go:526]: maybe? i look at this again, thanks for raising it.

Comment (sonnysideup) [reply pkg/buildkit/worker/pool.go:526]: @steved https://kubernetes.io/docs/reference/using-api/api-concepts/#semantics-for-get-and-list

> For **get**, the semantics of resourceVersion are: `resourceVersion` unset == "Most Recent"

Comment (steved) [reply pkg/buildkit/worker/pool.go:526]: Makes sense, I was just curious if we should be explicit. Thanks for checking.

File: pkg/controller/start.go (hunk 1 above)
Comment (sonnysideup) [reply pkg/controller/start.go:45]: We can just implement `MarshalJSON` for any of the configuration structs whenever we want to _sanitize_ rendered data.

**Pull request merged**